### PR TITLE
fixed bug #106 by minor changes Order, Orders Model and OrdersTest

### PR DESCRIPTION
**Pull request by asad041**
Fixed bug #106 by minor changes in Form/Order, Model/Orders and OrdersTest

### CODE REVIEW

File: app/Model/Orders.php [hunk 1]
```diff
@@ -46,15 +46,23 @@ public function orderIdExists($id, $userId)
         return $results[0]['cnt'] ? true : false;
     }
 
-    public function getActiveAmount($userId)
+    public function getActiveAmount($userId, $id = null)
```
Comment (thorie7912) [inline app/Model/Orders.php]: I think this should be `$excludeOrderId` to be clear. I didn't understand until I looked at more code.

File: tests/Model/OrdersTest.php [hunk 2]
```diff
@@ -225,6 +225,28 @@ public function testGetActiveAmount()
...
+            'maxAmount' => 120,
+            'status' => 0
+        ];
+
+        $sql = "select id, startDate, frequency, amount, maxAmount, status from orders where (id != ? AND userId = ? AND status = ?)";
+
+        $this->mockMysql->expectQuery($sql, [$row])
+            ->andNothingElse();
+
+        $orders->getActiveAmount(1, 1);
```
Comment (thorie7912) [inline tests/Model/OrdersTest.php]: We should assert the value being returned. What value are we expecting?

File: tests/Model/OrdersTest.php [hunk 3]
```diff
@@ -225,6 +225,28 @@ public function testGetActiveAmount()
         $orders->getActiveAmount(1);
     }
 
+    public function testGetActiveAmountHasOrderId()
```
Comment (thorie7912) [inline tests/Model/OrdersTest.php]: I think we also want add a separate test that verifies the maximum is not exceeded when excluding the order being modified. This was the original problem.

File: app/Model/Form/Order.php [hunk 4]
```diff
@@ -67,11 +67,13 @@ public function validate(array $params)
         } else {
             $this->calculator->addSchedule($params['startDate'], $params['frequency'], $params['amount'], $params['maxAmount']);
 
-            $activeOrders = $this->orders->getActiveAmount($params['userId']);
+            $editOrderId = empty($params['id']) ? null : $params['id'];
```
Comment (thorie7912) [inline app/Model/Form/Order.php]: We should validate `$params['id']` before using it. We should ensure that it is a number, no special characters, minimum and maximum length, and that an existing order exists with such a value.

File: app/Model/Form/Order.php [hunk 5]
```diff
@@ -67,11 +67,13 @@ public function validate(array $params)
         } else {
             $this->calculator->addSchedule($params['startDate'], $params['frequency'], $params['amount'], $params['maxAmount']);
```
Comment (thorie7912) [inline app/Model/Form/Order.php]: We should validate `$params['maxAmount']` before using it here.

File: .travis.yml [hunk 6]
```diff
@@ -1,7 +1,7 @@
 script:
   - export BRANCH=$(if [ "$TRAVIS_PULL_REQUEST" == "false" ]; then echo $TRAVIS_BRANCH; else echo $TRAVIS_PULL_REQUEST_BRANCH; fi)
   - echo "TRAVIS_BRANCH=$TRAVIS_BRANCH, PR=$PR, BRANCH=$BRANCH"
-  - curl -s "http://thorie.com/build.php?branch=$BRANCH" | tee /tmp/dock.out
+  - curl -s "http://thorie.com/build.php" -G --data-urlencode "branch=$BRANCH" | tee /tmp/dock.out
   - cat /tmp/dock.out | grep "@@SUCCESS@@"
-  - curl -s "http://thorie.com/build.php?branch=$BRANCH&phpver=7" | tee /tmp/dock.out
+  - curl -s "http://thorie.com/build.php?phpver=7" -G --data-urlencode "branch=$BRANCH" | tee /tmp/dock.out
```
Comment (thorie7912) [inline .travis.yml]: I changed this to fix the build error.

File: app/Model/Orders.php (hunk 1 above)
Comment (asad041) [reply app/Model/Orders.php]: I think it's necessary only if there's edit order form because if didn't provide then it will return current order record as well

File: app/Model/Form/Order.php (hunk 4 above)
Comment (asad041) [reply app/Model/Form/Order.php]: It's already validate with "OrderIdForm"

File: app/Model/Form/Order.php (hunk 5 above)
Comment (thorie7912) [reply app/Model/Form/Order.php]: I think we still call this when !$validMaxAmount so we should probably put the else part as an independent condition.

```
if ($validMaxAmount && $validStartDate && $validFrequency && $validAMount) { 
  addSchedule...
}
```
Does it make sense? Otherwise, it will setError() but continue to addSchedule().

File: app/Model/Orders.php (hunk 1 above)
Comment (thorie7912) [reply app/Model/Orders.php]: Yeah I'm just talking about renaming the variable to be more descriptive.

File: app/Model/Form/Order.php (hunk 5 above)
Comment (asad041) [reply app/Model/Form/Order.php]: Will split order Validate function in new branch so I think we can do this later!

**Comment by thorie7912** [comment]
Up to you, just don't forget
